plugin/trino-hive/src/main/java/io/trino/plugin/hive/s3select/S3SelectRecordCursorProvider.java

```diff
@@ -80,17 +90,20 @@ public Optional<ReaderRecordCursorWithProjections> createRecordCursor(
         // Ignore predicates on partial columns for now.
         effectivePredicate = effectivePredicate.filter((column, domain) -> column.isBaseColumn());
 
+        List<HiveColumnHandle> readerColumns = projectedReaderColumns
+                .map(readColumns -> readColumns.get().stream().map(HiveColumnHandle.class::cast).collect(toUnmodifiableList()))
```

Comment:
nit: we usually prefer the  `toImmutableList` collector from Guava

Reply:
Thanks, updating in the next commit.

Reply:
Done.

Reply:
Done.